README.md

```diff
@@ -5,7 +5,128 @@
 
 ## Getting started
 
-TODO
+### What does the Workleap.EventPropagation package do?
+* Provides list of existing Topics
+* Serializes domain events
+* Manages publication fields
+* Provides webhook endpoint (.net core only)
+* Provides IDomainEvent and IDomainEventHandler<> interfaces
+* Deserializes domain events
+* Mediates call to DomainEventHandlers
+* Validates configurations
+* Handles topic subscriptions
+
+### Using the Workleap.EventPropagation package to publish events
+
+When using dotnet core, you can register event propagation at startup in the service collection.
+```
+services
+    .AddEventPropagation()
+    .Configure(this.Configuration);
+```
+Configuration is required. Configuration can be loaded from the appsettings file by passing the IConfiguration instance (see above). The topic access key should be stored securely in a key vault.
+```
+"EventPropagation": {
+    "TopicName": "...",
+    "TopicAccessKey": "...",
+    "TopicEndpoint": "http://topicurl.domain.com/api/events"
+  }
+```
+
+To publish an event, use the IEventPropagationClient interface (via injection). Use the PublishDomainEventAsync to publish the event. The required Subject field is a string description to provide context for the event.
+
+```
+var coolCandorEvent = new ExampleDomainEvent
+{
+    CoolId = Guid.NewGuid().ToString(),
+    CoolDate = DateTime.UtcNow
+};
+
+await this._eventPropagationClient.PublishDomainEventAsync(subject: "TestEventPublication", coolCandorEvent);
+```
+
+#### Restrictions
+* Your service can publish events on only 1 topic.
+* The topic the service publishes to must be in the Topics class provided by the Workleap.EventPropagation package.
```

Comment:
Workleap.EventPropagation is not an existing package since ours is called DomainEventPropagation and I'm not sure if the whole sentence is still relevant or not

Reply:
This comment is still relevant, the whole sentence could be removed